Add back ProxyFix Middleware for flask app builder

gschuurman opened the pull request:
Fixes: #49781
Fixes: #49705

Add back the Removed ProxyFix Middleware to allow Oauth2 authentication when using a reverse proxy like nginx or traefik.

When not using proxyFix the redirect_url incorectly gets set to http:// instead of the expected https://, This breaks many authentication flows, especially azure, which does not allow http:// redirect_urls for anything other than localhost.

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

Commit:
Add back ProxyFix Middleware for flask app builder

Fixes Issues #49781 #49705

Add back the Removed ProxyFix Middleware to allow Oauth2 authentication
when using a reverse proxy like nginx or traefik.

When not using proxyFix the redirect_url incorectly gets set to http://
instead of the expected https://, This breaks many authentication flows,
especially azure, which does not allow http:// redirect_urls for
anything other than localhost.

Help is needed with adjusting the configuration from webserver to
api-server.

gschuurman committed:
Merge branch 'main' into feature/proxy_fix

vincbeck commented on providers/fab/src/airflow/providers/fab/www/extensions/init_wsgi_middlewares.py:
```diff
@@ -0,0 +1,42 @@
+    """Handle X-Forwarded-* headers and base_url support."""
+
+    # Apply ProxyFix middleware
+    if conf.getboolean("webserver", "ENABLE_PROXY_FIX"):
```
We removed this config because it was no longer used. Please update `airflow-core/src/airflow/cli/commands/config_command.py` as well to no longer mark this config as removed.

vincbeck replied:
I think it would be better to move this config to `fab` instead of keeping it in `webserver` but I think it is too late since we already cut 3.0.0. Unless we consider it a bug fix and put it in 3.0.X branch, then we could still rename this config? Maybe @potiuk knows better on that question

gschuurman replied:
Completly agree, I moved the configuration from webserver to fab and added the documentation in the provider docs.

Commit:
Add config documentation to fab provider

Add the original proxy_fix documentation back into the FAB provider
package and update the section to match fab instead of websever.

eladkal commented on providers/fab/provider.yaml:
```diff
@@ -119,6 +119,58 @@ config:
+      enable_proxy_fix:
+        description: |
+          Enable werkzeug ``ProxyFix`` middleware for reverse proxy
+        version_added: 3.0.1
```
versions are provider versions not Airflow core versions
it should be 2.1.0

Commit:
Use correct versioning scheme for provider

Commit:
Fix typo in import

Commit:
Bypass missing extention test
